### PR TITLE
[explorer/nodewatch]: various fixes and updates

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: explorer/nodewatch/nodewatch/__init__.py [hunk 1]
```diff
@@ -39,12 +39,16 @@ def nem_summary():  # pylint: disable=unused-variable
...
+
+	@app.route('/api/nem/chart/height')
+	def api_nem_chart_height():  # pylint: disable=unused-variable
 		return nem_routes_facade.json_height_chart_with_metadata()
 
-	@app.route('/nem/height')
-	def nem_height():  # pylint: disable=unused-variable
+	@app.route('/api/nem/height')
+	def api_nem_height():  # pylint: disable=unused-variable
 		return nem_routes_facade.json_height()
```
Comment (gimre-xymcity) [inline explorer/nodewatch/nodewatch/__init__.py:52]: should we keep rendered vs non-rendered apis in some kind of separate groups/regions?

File: explorer/nodewatch/nodewatch/NetworkRepository.py [hunk 2]
```diff
@@ -12,19 +12,52 @@
...
 		self.main_address = main_address
+		self.main_public_key = main_public_key
 		self.endpoint = endpoint
 		self.name = name
 		self.version = version
 		self.height = height
 		self.finalized_height = finalized_height
 		self.balance = balance
-		self.has_api = has_api
+		self.roles = roles
```
Comment (gimre-xymcity) [inline explorer/nodewatch/nodewatch/NetworkRepository.py:40]: not sure if we need to add this now, but I guess adding 'node' key might be handy (for node key link txes)

File: explorer/nodewatch/nodewatch/NetworkConnector.py [hunk 3]
```diff
@@ -62,7 +65,7 @@ async def _update_height(self, session, descriptor):
 
 				if 'finalized_height' in response_json:
 					descriptor.finalized_height = int(response_json['finalized_height'])
-		except (aiohttp.client_exceptions.ClientConnectorError, asyncio.TimeoutError) as ex:
+		except (aiohttp.client_exceptions.ClientConnectorError, aiohttp.client_exceptions.ContentTypeError, asyncio.TimeoutError) as ex:
```
Comment (gimre-xymcity) [inline explorer/nodewatch/nodewatch/NetworkConnector.py:68]: why there's ContentType exception?

File: explorer/nodewatch/nodewatch/__init__.py (hunk 1 above)
Comment (Jaguar0625) [reply explorer/nodewatch/nodewatch/__init__.py:52]: it should be already

/nem
/api/nem
/symbol
/api/symbol

do you mean to swap /api/nem and /symbol ?

File: explorer/nodewatch/nodewatch/NetworkConnector.py (hunk 3 above)
Comment (Jaguar0625) [reply explorer/nodewatch/nodewatch/NetworkConnector.py:68]: was hitting it yesterday when nodes were stopped/updating but webserver was still running and returning text/html 502 page 😬

File: explorer/nodewatch/nodewatch/NetworkRepository.py (hunk 2 above)
Comment (Jaguar0625) [reply explorer/nodewatch/nodewatch/NetworkRepository.py:40]: can add it

Comment (Jaguar0625) [reply explorer/nodewatch/nodewatch/NetworkRepository.py:43]: notice this is not in normal peers information
and currently. due to way ingestion script, works it is not being retrieved for peer-only nodes

File: explorer/nodewatch/nodewatch/__init__.py (hunk 1 above)
Comment (gimre-xymcity) [reply explorer/nodewatch/nodewatch/__init__.py:55]: I thought about completely separating, as we have:
/nem/harvesters
/nem/nodes
...

/symbol/voters
/symbol/harvesters
...

keep apis completely separate down in
/api/nem/foo
/api/symbol/foo

not sure if good idea, just an idea

File: explorer/nodewatch/nodewatch/NetworkRepository.py (hunk 2 above)
Comment (gimre-xymcity) [reply explorer/nodewatch/nodewatch/NetworkRepository.py:43]: hmm, you mean peer-only or having peer role? (the latter one would be fine ofc 😁 )

**Review by gimre-xymcity — APPROVED** [review]

File: explorer/nodewatch/nodewatch/NetworkRepository.py (hunk 2 above)
Comment (Jaguar0625) [reply explorer/nodewatch/nodewatch/NetworkRepository.py:43]: peer only

File: explorer/nodewatch/nodewatch/__init__.py (hunk 1 above)
Comment (Jaguar0625) [reply explorer/nodewatch/nodewatch/__init__.py:55]: i'd rather separate nem / symbol and then api within that

**Pull request merged**